### PR TITLE
remove createNewUser, use a direct call to prisma

### LINKED ISSUE #150: Remove createNewUser() function
Since we our now hashing password within Prisma's middleware, this function is not doing anyting, except wrapping a call of `prisma.user.create()`. We can remove it and replace it with direct invocations of prisma client.

### PR DESCRIPTION
**Pull request by thvu11**
# Description

 Remove `createNewUser` as the function is no longer used and it is replaced with a direct call to prisma.

Fixes #150

## Type of change

Please select the option(s) that is more relevant.

- [ x] Code cleanup
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by Martinsos** [comment]
@thvu11 thanks for creating the PR!
I think you did refactoring correctly, but I believe one thing is missing: you need to make sure that appropriate template data is provided to the signup.js template, in this case that is userEntityLower. However, if you take a look at waspc/src/Generator/ServerGenerator/AuthG.hs, you will notice that genSignupRoute, which generates signup.js file, does not provide any template data, it just copies the template file with no provided data.
This will unfortunately compile, since we are not capturing this stuff in types for now.

What you should do is, modify `genSignupRoute` to look like `genLoginRoute`, although you will need only userEntityLower, and will not need userEntityUpper.
Let me know if you need any help with this! The whole "filedraft" system that we have come up with might be somewhat confusing at first.

What about testing -> how have you tested the code? I guess you compiled it? In this case, you should try running an example app and checking that signup works. Let me know if this is a problem for some reason and I can also try to help with that!

**Review by Martinsos — COMMENTED** [review]
Looking good, just missing one thing, I left the comment!

**Comment by thvu11** [comment]
Thanks for the advice, I completely miss that. It seems like this happened as my workflow is still a bit glitchy. I was doing `run ghcid` on one tab to watch the project and automatically rebuild it on any changes while constantly doing `stack exec wasp start` inside `example/todoApp` directory. I'm still a bit unsure why the change on the Haskell code didn't show up in the example app.

**Comment by Martinsos** [comment]
@thvu11 No problem, this is a good opportunity to help you figuring out that workflow!
You fixed that correctly, but if you run TodoApp with `stack exec wasp start`, right now it will fail.
What is still missing is `{{={= =}=}}` at the top of `signup.js` file -> it tells mustache (which is template engine we use) that template tags we use are `{=` and `=}` -> once you add that, we can merge.
But first, make sure to get TodoApp to fail (React in browser should throw error).

What you are probably missing is `stack build`! The thing is, if I am correct, that `ghcid` does not build the project for "real" -> what it does is load all modules into a `ghci` session and then it works with that. But when you run `stack exec wasp start`, it looks for built/compiled code that was previously compiled/built by `stack built` -> `ghcid` works separately.
TLDR: before running `stack exec wasp start`, you need to run `stack build` to compile the latest code, regardless of `ghcid`.

**Comment by thvu11** [comment]
Great! Thank you for the instructions, I finally make the workflow run smoothly.

**Comment by Martinsos** [comment]
All right that is it, awesome! Merging :).
